packages/engine/src/worker/mod.rs

```diff
@@ -285,7 +353,7 @@ impl WorkerController {
         Ok(())
     }
 
-    /// TODO: DOC
+    /// Attempts to join the spawned runners.
     async fn terminate_runners(&mut self) -> Result<()> {
         tokio::try_join!(
```

Comment:
can you add:
// TODO: these methods don't look like they need to be async

Reply:
I think they're async for future-proofing, e.g. if we discover that using nng in a different way gives better performance, but requires making something async.

Reply:
I'm not sure how marking them as async future-proofs it though. Surely at that point we could just convert them to async and make changes as necessary.

I think in the mean-time we should remove async where it's unnecessary. I've done some googling and it seems that using async for synchronous code adds negligible overhead, but I'm not sure that's necessarily true if you put things like `try_join_all`. And even then it adds development overhead. There's a task open for it though, so lets discuss when that task is picked up :)